tests/UI/pages/BO/catalog/productsV2/add/combinationsTab.js

```diff
@@ -89,6 +202,421 @@ class CombinationsTab extends BOBasePage {
+
+    return this.getPaginationLabel(page);
+  }
+
+  // Methods for filter combinations by size
+  /**
+   * Get number of combinations displayed in list
+   * @param page {Page} Browser tab
+   * @returns {Promise<number>}
+   */
+  async getNumberOfCombinationsFromList(page) {
+    const footerText = await this.getTextContent(page, this.paginationLabel);
+    const numberOfCombinations = /\d+/g.exec(footerText.match(/out of ([0-9]+)/)).toString();
+
+    return parseInt(numberOfCombinations, 10);
+  }
+
+  /**
+   * Filter combinations by color
+   * @param page {Page} Browser tab
+   * @param sizeID {number} Size number in list
+   * @returns {Promise<void>}
+   */
+  async filterCombinationsBySize(page, sizeID) {
+    await this.waitForSelectorAndClick(page, this.filterBySizeButton);
+    await this.waitForVisibleSelector(page, `${this.filterBySizeDropDownMenu}.show`);
+    await this.setChecked(page, this.filterBySizeCheckboxButton(sizeID));
+
+    await this.waitForSelectorAndClick(page, this.filterBySizeButton);
+  }
+
+  /**
+   * Get filter button name
+   * @param page {Page} Browser tab
+   * @returns {Promise<string>}
+   */
+  getFilterBySizeButtonName(page) {
+    return this.getTextContent(page, this.filterBySizeButton);
+  }
+
+  /**
+   * Clear filter
+   * @param page {Page} Browser tab
+   * @returns {Promise<number>}
+   */
+  async clearFilter(page) {
+    await this.waitForSelectorAndClick(page, this.clearFilterButton);
+
+    return this.getNumberOfCombinationsFromList(page);
+  }
+
+  // Methods for bulk actions
+  /**
+   * Select All combinations
+   * @param page {Page} Browser tab
+   * @param allCombinations {boolean} true to select all combinations, false to select all in page
+   * @returns {Promise<boolean>}
+   */
+  async selectAllCombinations(page, allCombinations = true) {
+    await this.waitForSelectorAndClick(page, this.combinationListTableSelectAllButton);
+
+    await this.waitForVisibleSelector(page, this.combinationListSelectAllDropDownMenu);
+    if (allCombinations) {
+      await this.setChecked(page, this.combinationListBulkSelectAll);
+    } else await this.setChecked(page, this.combinationListBulkSelectAllInPage);
+
+    return this.elementVisible(page, this.bulkActionsButton, 2000);
+  }
+
+  /**
+   * Click on edit Combinations by bulk actions
+   * @param page {Page} Browser tab
+   * @returns {Promise<string>}
+   */
+  async clickOnEditCombinationsByBulkActions(page) {
+    await this.waitForSelectorAndClick(page, this.bulkActionsButton);
+    await this.waitForSelectorAndClick(page, this.bulkEditButton);
+
+    await this.waitForVisibleSelector(page, this.bulkEditModal);
+
+    return this.getTextContent(page, this.bulkEditModalTitle);
+  }
+
+  /**
+   * Edit combinations by bulk actions
+   * @param page {Page} Browser tab
+   * @param editStockData {object}
+   * @returns {Promise<string>}
+   */
+  async editCombinationsByBulkActions(page, editStockData) {
```

Comment:
I think this function could be slip on multiple one, so it will be clear to use

Reply:
done :+1: